app/src/main/java/infinite_zoomer/model/DrawingModel.java

```diff
@@ -4,20 +4,49 @@
  */
 
 package infinite_zoomer.model;
+import infinite_zoomer.gui.AbstractGUI;
 import infinite_zoomer.model.geometry.Circle;
 import infinite_zoomer.model.geometry.Rectangle;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Consumer;
 
 
 /**
  * Internal representation of the drawing (data on the canvas).
  */
 public class DrawingModel {
     private ContainerNode mRoot;
+    private updateListener listener;
 
     public DrawingModel() {
         mRoot = new ContainerNode();
+        listener = new updateListener();
+    }
+
+    //updates GUI's when called
+    //TODO: resolve if it should be static or not (?)
+    public class updateListener {
```

Comment:
```suggestion
    private UpdateListener listener;

    public DrawingModel() {
        mRoot = new ContainerNode();
        listener = new updateListener();
    }

    //updates GUI's when called
    //TODO: resolve if it should be static or not (?)
    public class UpdateListener {
```

Let's use CamelCase here!
Also, I think `updateListener` can be static &mdash; it doesn't need to access any of `DrawingModel`'s internal data. [Making it static makes it independent of any particular instance of `DrawingModel`](https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html). It can be independent, so let's make it independent!